### ISSUE TITLE
[docs] Mega tracking: improve quickstart, training docs, and agent readiness

### ISSUE THREAD

**Issue by yaoyu-33** [NVIDIA-NeMo/Megatron-Bridge#3814]
## Summary

External onboarding notes show that the current documentation is broadly useful and several core workflows work as expected, but users still have to infer too much around checkpoint conversion, training entry points, data preparation, and stale example paths.

This tracker collects the documentation and onboarding work needed to make Megatron Bridge easier to adopt for HF -> Megatron conversion, random-init pretraining from HF architecture, SFT/PEFT from pretrained checkpoints, precision workflows, and agent-driven development.

Linear tracker: https://linear.app/nvidia/issue/MB-358/docs-mega-tracking-improve-quickstart-training-docs-and-agent

## Baseline positives to preserve

- Quickstart snippets for HF checkpoint conversion to Megatron format and launching pretraining from an HF checkpoint work as expected.
- Megatron Bridge can initialize a Hugging Face model architecture and pretrain from random weights without requiring a prior Megatron-format checkpoint.
- Full finetuning and PEFT workflows work after converting HF checkpoints into Megatron format.
- Training docs have useful navigation, with `training/README.md` acting as a workflow hub.
- Distributed training and performance docs cover parallelism, mixed precision, communication overlap, activation recomputation, and profiling in useful detail.
- Performance skills use exact source references, and the structured YAML metadata beside skills is useful for machine-readable validation and tracking.

## Quickstart work

- [x] Clarify whether the HF -> Megatron conversion script is required, optional, or workflow-specific. Explicitly call out that random-init pretraining from an HF architecture does not require a converted Megatron checkpoint.
- [x] In the pre-configured recipe quickstart, show how to supply each supported model source: Hugging Face model ID, local HF model path, and converted Megatron checkpoint.
- [x] Add a concrete recipe snippet for loading an HF architecture without weights, for example:

```python
cfg.model = AutoBridge.from_hf_pretrained("meta-llama/Llama-3.2-1B").to_megatron_provider(load_weights=False)
```

- [x] Link the repository `tutorials/` directory directly from the Quick Start path so users discover data preparation, recipes, and training examples earlier.

## Training workflow work

- [x] Clarify the distinction between `pretrain` and `finetune`, including whether PEFT configs are intentionally supported through `pretrain` or whether the docs/API should steer users elsewhere.
- [x] Document the default LoRA finetuning checkpoint requirement. Make clear whether `pretrained_checkpoint` must point to a converted Megatron checkpoint, or whether an HF model ID/path is accepted.
- [x] Fix the malformed LoRA computation equation rendering in the PEFT documentation.
- [x] Add a getting-started section for NVFP4 pretraining and finetuning, including how it differs from or relates to FP8 workflows.
- [x] Add guidance for deciding whether to reuse an existing recipe config or create a new config when training an HF model.
- [x] Clarify checkpoint resume behavior. Users observed optimizer shape mismatch after a pretraining run because later runs implicitly loaded checkpoints and optimizer state from `nemo_experiments`. Decide whether this should default to a fresh run unless a checkpoint is explicitly provided, or document the current behavior very prominently.
- [x] State the conversion requirement by workflow: random-init pretraining can load the HF architecture without a Megatron checkpoint, while SFT/PEFT workflows requiring `pretrained_checkpoint` need a Megatron-format checkpoint unless another source is explicitly supported.
- [x] Add a dedicated data preparation guide covering Megatron binary format for LLM pretraining, JSONL schemas for SFT/PEFT, WebDataset/Energon formats for VLM fine-tuning, and Hugging Face dataset integration.

## Documentation correctness work

- [x] Fix the `pretrain_config` import error in `docs/recipe-usage.md`; the referenced module path does not exist.
- [x] Normalize guidance for `pretrained_checkpoint`: state whether it should point to the base checkpoint directory or a specific `iter_N` subdirectory, and explain when the latest checkpoint is loaded versus an intermediate iteration.
- [x] Normalize config naming across examples and implementation. Known inconsistencies include `data_path` versus `dataset_root`, and `sequence_length` versus `seq_length`.

## Agentic readiness work

- [x] Add or document a root-level agent entry file, such as `AGENTS.md`, `CLAUDE.md`, or equivalent, so coding agents can discover and use the repository skills automatically.
- [x] Fix CUDA graphs performance documentation that references `scripts/performance/run_performance_workload.py`; that entry point does not exist. Update the docs to the actual entry point.
- [x] Fix CUDA graphs verification instructions that point to the wrong test path.
- [x] Fix model support documentation that references an FP8 dequantization script at a nonexistent path.

## Acceptance criteria

- [x] A new user can follow the Quick Start and understand exactly when checkpoint conversion is needed and when it is not.
- [x] The docs contain explicit examples for HF model IDs, local HF paths, and converted Megatron checkpoints.
- [x] SFT/PEFT docs make the expected checkpoint format and `pretrained_checkpoint` path semantics unambiguous.
- [x] Data preparation docs describe expected input formats and point to runnable examples.
- [x] Precision docs include an NVFP4 entry path alongside FP8 guidance.
- [x] Referenced scripts, imports, test paths, and config names are validated against the repository.
- [x] Agent-facing repository entry points make the existing `skills/` directory discoverable.

## Notes

Several source notes included link placeholders (`link`) without actual URLs. Attach the original repro links or source references as follow-up comments when available.

Some items may split naturally into child issues, especially checkpoint resume semantics, data preparation docs, NVFP4 docs, and agent-readiness path fixes.
